src/Nethermind/Nethermind.State/VerkleStateTree.cs

```diff
@@ -70,6 +71,36 @@ public void SetCode(Address address, byte[] code)
             Insert(key, chunk);
             chunkId += 1;
         }
+
+        // Update code size
+        AccountStruct? account = Get(address);
+        if (account.HasValue)
+        {
+            var newAccount = new Account(account.Value.Nonce, account.Value.Balance, (UInt256)code.Length, account.Value.Version, new Hash256(account.Value.StorageRoot), Keccak.Compute(code));
+            Set(address, newAccount);
+        }
```

Comment:
dont add this here - the way this is used in client - we update the code size and then call SetCode()
so you should move it outside